tests/transforms/test_split_to_single_terms.py

```diff
@@ -0,0 +1,661 @@
+        assert np.allclose(fn(([0.1, 0.2],)), 0.3)
+
+    def test_multiple_sums(self):
+        """Test that the transform works as expected for a circuit that
+        returns multiple sums"""
+        tape = qml.tape.QuantumScript(
+            [], [qml.expval(qml.X(0) + qml.Y(1)), qml.expval(qml.X(2) + qml.Z(3))]
+        )
+        tapes, fn = split_to_single_terms(tape)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [
+            qml.expval(qml.X(0)),
+            qml.expval(qml.Y(1)),
+            qml.expval(qml.X(2)),
+            qml.expval(qml.Z(3)),
+        ]
+        assert fn(([0.1, 0.2, 0.3, 0.4],)) == (0.1 + 0.2, 0.3 + 0.4)
+
+    def test_multiple_sums_overlapping(self):
+        """Test that the transform works as expected for a circuit that
+        returns multiple sums, where some terms are included in multiple sums"""
+        tape = qml.tape.QuantumScript(
+            [], [qml.expval(qml.X(0) + qml.Y(1)), qml.expval(qml.X(2) + qml.Y(1))]
+        )
+        tapes, fn = split_to_single_terms(tape)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [
+            qml.expval(qml.X(0)),
+            qml.expval(qml.Y(1)),
+            qml.expval(qml.X(2)),
+        ]
+        assert fn(([0.1, 0.2, 0.3],)) == (0.1 + 0.2, 0.3 + 0.2)
+
+    def test_multiple_sums_duplicated(self):
+        """Test that the transform works as expected for a circuit that returns multiple sums, where each
+        sum includes the same term more than once"""
+        tape = qml.tape.QuantumScript(
+            [], [qml.expval(qml.X(0) + qml.X(0)), qml.expval(qml.X(1) + qml.Y(1) + qml.Y(1))]
+        )
+        tapes, fn = split_to_single_terms(tape)
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [
+            qml.expval(qml.X(0)),
+            qml.expval(qml.X(1)),
+            qml.expval(qml.Y(1)),
+        ]
+        assert fn(([0.1, 0.2, 0.3],)) == (0.1 + 0.1, 0.2 + 0.3 + 0.3)
+
+    @pytest.mark.parametrize("batch_type", (tuple, list))
+    def test_batch_of_tapes(self, batch_type):
+        """Test that `split_to_single_terms` can transform a batch of tapes with multi-term observables"""
+
+        tape_batch = [
+            qml.tape.QuantumScript([qml.RX(1.2, 0)], [qml.expval(qml.X(0) + qml.Y(0) + qml.X(1))]),
+            qml.tape.QuantumScript([qml.RY(0.5, 0)], [qml.expval(qml.Z(0) + qml.Y(0))]),
+        ]
+        tape_batch = batch_type(tape_batch)
+
+        tapes, fn = split_to_single_terms(tape_batch)
+
+        expected_tapes = [
+            qml.tape.QuantumScript(
+                [qml.RX(1.2, 0)],
+                [qml.expval(qml.X(0)), qml.expval(qml.Y(0)), qml.expval(qml.X(1))],
+            ),
+            qml.tape.QuantumScript([qml.RY(0.5, 0)], [qml.expval(qml.Z(0)), qml.expval(qml.Y(0))]),
+        ]
+        for actual_tape, expected_tape in zip(tapes, expected_tapes):
+            assert qml.equal(actual_tape, expected_tape)
```

Comment:
```suggestion
            qml.assert_equal(actual_tape, expected_tape)
```

While provide too much information right now, it will eventually raise an error to explain why they aren't equal, which will make the test easier to debug if it ever fails.